src/Microsoft.ComponentDetection.Detectors/npm/NpmComponentDetector.cs

```diff
@@ -81,16 +84,80 @@ protected virtual bool ProcessIndividualPackageJTokens(string filePath, ISingleF
+            
             if (!SemanticVersion.TryParse(version, out _))
             {
                 Logger.LogWarning($"Unable to parse version \"{version}\" for package \"{name}\" found at path \"{filePath}\". This may indicate an invalid npm package component and it will not be registered.");
                 return false;
             }
 
-            var detectedComponent = new DetectedComponent(new NpmComponent(name, version));
+            var npmComponent = new NpmComponent(name, version);
+            npmComponent.Author = GetAuthor(authorToken, name, filePath);
+
+            var detectedComponent = new DetectedComponent(npmComponent);
+            
             singleFileComponentRecorder.RegisterUsage(detectedComponent);
             return true;
         }
+
+        private NpmAuthor GetAuthor(JToken authorToken, string packageName, string filePath)
+        {
+            var authorString = authorToken?.ToString();
+            if (string.IsNullOrEmpty(authorString))
+            {
+                return null;
+            }
+
+            string authorName = null;
+            string authorEmail = null;
+
+            //for parsing author in Json Format
+            // for e.g. 
+            // "author": {
+            //     "name": "John Doe",
+            //     "email": "johndoe@outlook.com",
+            //     "name": "https://jd.com",
+            // }
+            if (authorString.StartsWith("{"))
+            {
+                authorName = authorToken["name"]?.ToString();
+                authorEmail = authorToken["email"]?.ToString();
+            }
+
+            // for parsing single line author.
+            // for e.g. "author": "John Doe <johnDoe@outlook.com> (https://jd.com)"
+            else
+            {
+                bool isEmailPresent = authorString.Contains("<");
+                bool isUrlPresent = authorString.Contains("(");
+                if (!isEmailPresent)
+                {
+                    if (isUrlPresent)
+                    {
+                        authorName = authorString.Substring(0, authorString.IndexOf("(")).Trim();
+                    }
+                    else
+                    {
+                        authorName = authorString.Trim();
+                    }
+                }
+                else
+                {
+                    authorName = authorString.Substring(0, authorString.IndexOf("<")).Trim();
+                    authorEmail = authorString.Substring(authorString.IndexOf("<") + 1, authorString.IndexOf(">") - authorString.IndexOf("<") - 1);
+                }
+            }
+
+            if (string.IsNullOrEmpty(authorName))
+            {
+                Logger.LogWarning(string.Format(
+                    "Unable to parse author: {0} for package: {1} found at path {2}. " +
+                    "This may indicate an invalid npm package author, and it will not be registered.", authorString, packageName, filePath));
```

Comment:
nit:
```suggestion
                Logger.LogWarning(
                    $"Unable to parse author: {authorString} for package: {packageName} found at path {filePath}. This may indicate an invalid npm package author, and it will not be registered.");
```